apps/condo/domains/miniapp/schema/AppMessageSetting.test.js

```diff
@@ -0,0 +1,429 @@
+                })
+
+                test('can\'t delete', async () => {
+                    const [appMessageSetting] = await createTestAppMessageSetting(admin, {
+                        b2cApp,
+                    })
+
+                    await expectToThrowAccessDeniedErrorToObj(async () => {
+                        await AppMessageSetting.delete(support, appMessageSetting.id)
+                    })
+                })
+
+                test('can read', async () => {
+                    const [obj] = await createTestAppMessageSetting(support, {
+                        b2cApp,
+                    })
+
+                    const appSetting = await AppMessageSetting.getOne(support, {
+                        id: obj.id,
+                    })
+
+                    expect(appSetting).toBeDefined()
+                    expect(appSetting.id).toEqual(obj.id)
+                    expect(appSetting.type).toEqual(B2C_APP_MESSAGE_PUSH_TYPE)
+                })
+            })
+
+            describe('User', () => {
+                test('can\'t create', async () => {
+                    await expectToThrowAccessDeniedErrorToObj(async () => {
+                        await createTestAppMessageSetting(user, { b2cApp })
+                    })
+                })
+
+                test('can\'t update', async () => {
+                    const [appMessageSetting] = await createTestAppMessageSetting(admin, {
+                        b2cApp,
+                    })
+
+                    await expectToThrowAccessDeniedErrorToObj(async () => {
+                        await updateTestAppMessageSetting(user, appMessageSetting.id)
+                    })
+                })
+
+                test('can\'t delete', async () => {
+                    const [appMessageSetting] = await createTestAppMessageSetting(admin, {
+                        b2cApp,
+                    })
+
+                    await expectToThrowAccessDeniedErrorToObj(async () => {
+                        await AppMessageSetting.delete(user, appMessageSetting.id)
+                    })
+                })
+
+                test('can\'t read', async () => {
+                    const [appMessageSetting] = await createTestAppMessageSetting(admin, {
+                        b2cApp,
+                    })
+
+                    const readAppMessageSetting = await AppMessageSetting.getOne(user, {
+                        id: appMessageSetting.id,
+                    })
+
+                    expect(readAppMessageSetting).toBeUndefined()
+                })
+
+                test('user with employee with b2bAppRole in organization with b2b app from setting can read', async () => {
+                    const staffUser = await makeClientWithNewRegisteredAndLoggedInUser()
+
+                    const [organization] = await createTestOrganization(admin)
+                    const [app] = await createTestB2BApp(admin)
```

Comment:
Try to reduce amount of `admin` calls in test files, so your tests are close to prod. 

By doing it, you're not just testing your logic directly, but instead testing a bunch of stuff implicitly, so your tests become more useful